zubhub_frontend/zubhub/src/views/create_project/CreateProject.jsx

```
@@ -0,0 +1,684 @@
+      props.setFieldTouched("materials_used");
+
+      props.validateField("title");
+      props.validateField("description");
+      props.validateField("materials_used");
+
+      let media_fields = mediaFieldsValidation();
+
+      if (media_fields.image_is_empty && media_fields.video_is_empty) {
+        return;
+      } else if (media_fields.video_value_not_url) {
+        return;
+      } else if (media_fields.too_many_images === true) {
+        return;
+      } else if (media_fields.image_size_too_large === true) {
+        return;
+      } else if (media_fields.image_is_empty) {
+        upload_project();
+      } else {
+        let project_images = document.querySelector("#project_images").files;
+
+        let { image_upload } = state;
+        image_upload.images_to_upload = project_images.length;
+        image_upload.upload_dialog = true;
+        image_upload.upload_percent = 0;
+        setState({ ...state, image_upload });
+
+        for (let index = 0; index < project_images.length; index++) {
+          upload(project_images[index]);
+        }
+      }
+    }
+  };
+
+  const mediaFieldsValidation = () => {
+    let image_upload_button = document.querySelector("#image_upload_button");
+    let media_fields = document.querySelector("#project_images");
+    let video = document.querySelector("#video");
+    let imageCount = document.querySelector(".imageCountStyle");
+    imageCount.innerText = media_fields.files.length;
+    imageCount.style.fontSize = "0.8rem";
+
+    let result = {};
+
+    if (/^(ftp|http|https):\/\/[^ "]+$/.test(video.value) === false) {
+      props.setFieldTouched("video");
+      props.setFieldError(
+        "video",
+        "you are required to submit a video url here"
+      );
+      result["video_value_not_url"] = true;
+    }
+
+    if (media_fields.files.length < 1) {
+      result["image_is_empty"] = true;
+      if (video.value === "") {
+        image_upload_button.setAttribute(
```

I am not sure if the styles will be overridden or not if a re-render happens in the component

---

they will be overridden, but that is the point. form errors are not to be persisted if it can be avoided since it changes on every validation call

---

Form errors should not persist but their might be a chance that the errors are lost before user completely reading the errors because of a re-render